src/OpenTracingMock/Tracer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace OpenTracingMock;
+
+use OpenTracing\Exceptions\UnsupportedFormat;
+use OpenTracing\SpanOptions;
+use OpenTracing\Tracer as OTTracer;
+use OpenTracing\SpanContext as OTSpanContext;
+
+final class Tracer implements OTTracer
+{
+    /**
+     * @var array|Span[]
+     */
+    private $spans = [];
+
+    /**
+     * @var array|callable[]
+     */
+    private $injectors;
+
+    /**
+     * @var array|callable[]
+     */
+    private $extractors;
+
+    private function __construct(array $injectors, array $extractors)
+    {
+        $this->injectors = $injectors;
+        $this->extractors = $extractors;
+    }
+
+    public static function create(array $injectors = [], array $extractors = [])
+    {
+        return new self($injectors, $extractors);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function startSpan($operationName, $options = [])
```

Comment:
Method should return a span as per the spec

Reply:
It's a bit concerning seeing comments like this - do we not have a test harness like we have in Python / NodeJS? We even have one for Go, but it's especially important for dynamic languages.

Reply:
Totally, this was a WIP so there were many things missing. I just uploaded all the changes and fixed most of the details.